src/Microsoft.AspNetCore.SignalR.Core/Internal/HubDispatcher.cs

```diff
@@ -0,0 +1,461 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Internal
+{
+    public partial class HubDispatcher<THub> : IInvocationBinder where THub : Hub
```

Comment:
Is this implementing IInvocationBinder an implementation requirement?

Reply:
It is right now. The Invocation Binder and the Dispatcher both need to share the metadata about the methods in the Hub. We could do that in two ways: Extract "Discovery" into a separate component and have both the Binder and Dispatcher use it, or make the Binder and Dispatcher the same component and do discovery internally. Until we decide what to do with Discovery, this seemed like the most straightforward way to do it.